snap/hooktypes.go

```diff
@@ -26,6 +26,7 @@ import (
 var supportedHooks = []*HookType{
 	newHookType(regexp.MustCompile("^prepare-device$")),
 	newHookType(regexp.MustCompile("^configure$")),
+	newHookType(regexp.MustCompile("^prepare-(?:plug|slot)-[a-zA-Z0-9_\\-]+$")),
```

Comment:
I agree about the settle thing in the test being ugly, but the in fact it is used in a couple of places already; I'm happy to discuss this of course, but perhaps it should be addressed in a separate PR.

~/gocode/src/github.com/snapcore/snapd/overlord $ grep -r -C1 "i < 50" *
assertstate/assertmgr_test.go-  // XXX: would like to use Overlord.Settle but not enough control there
assertstate/assertmgr_test.go:  for i := 0; i < 50; i++ {
assertstate/assertmgr_test.go-          s.mgr.Ensure()
--
devicestate/devicemgr_test.go-func (s *deviceMgrSuite) settle() {
devicestate/devicemgr_test.go:  for i := 0; i < 50; i++ {
devicestate/devicemgr_test.go-          s.hookMgr.Ensure()
--
snapstate/snapmgr_test.go-      // FIXME: use the real settle here
snapstate/snapmgr_test.go:      for i := 0; i < 50; i++ {
snapstate/snapmgr_test.go-              s.snapmgr.Ensure()
--
snapstate/booted_test.go-func (bs *bootedSuite) settle() {
snapstate/booted_test.go:       for i := 0; i < 50; i++ {
snapstate/booted_test.go-               bs.snapmgr.Ensure()